### DISCUSSION TITLE
Completed achievements as a condition

### DISCUSSION THREAD

**Discussion by Jamiras** [RetroAchievements/RAIntegration#768] [Ideas]
As a developer, I'd like to create an achievement that requires one or more other achievements to be completed first.

For example, in Mario Kart, I might have achievements for getting gold in Mushroom Cup, Flower Cup, Star Cup, and Special Cup. Then I'd have a meta-achievement for getting the gold in the other four cups. Since the other achievements can be earned independently, the meta-achievement can be earned across multiple sessions.

The implementation would be to change the requirement to be "Achievement 12345 = 1". Which would serialize to something like "@12345=1". Note the new "@" prefix to differentiate it from Delta, Memory and Value types. Something to consider is how to represent this for local development where achievements all have ID=0.

**Comment by Jamiras** [comment]
It has also been proposed that the sub-achievements be non-visible and serve as multi-session checkpoints, in which case the sub-achievement would be worth 0 points. There are probably better ways to implement multi-session checkpoints.

The toolkit would have to have some way to flag an achievement as being non-visible (maybe just having it worth 0 points would be sufficient?)

It could be desirable to allow the player to see which hidden sub-achievements they've earned, perhaps using a folder-like interface (see #769)

**Comment by meleu** [comment]
> Something to consider is how to represent this for local development where achievements all have ID=0.

Since we changed the behavior of Unofficial achievements in https://github.com/RetroAchievements/RAWeb/pull/141, we can safely upload WIP cheevos to Unofficial and use their ID.

**Comment by wilhitewarrior** [comment]
I talked about this recently with some of the other guys, I think this would be an awesome feature

**Comment by NIWDERED07** [comment]
This is a great idea, has there been any progress on it?
